draft-secure-credential-transfer.md

```diff
@@ -129,69 +140,67 @@ Receiver device, having read the encrypted Provisioning Information from the Rel
+The process starts with a Sender device building and encrypting Provisioning Information with the Secret and storing it on a Relay server in a unique mailbox. A unique MailboxIdentifier is generated by the Sender device, created from random data with good source of entropy (preferrably hardware-based entropy).
 
-In addition to encrypted Provisioning Information, Sender stores a Sender Notification Token in the mailbox, which allows Relay server to send a notification message to the Sender device. Sender device generates a unique token - a Sender Device Claim - and stores it to the mailbox. Device Claim allows the Sender device presenting it to read and write data to/from the mailbox, thus binding it to the mailbox.
+In addition to encrypted Provisioning Information, Sender stores a Sender Notification Token in the mailbox, which allows Relay server to send a notification to the Sender device. Sender device generates a unique token - a Sender Device Claim - and stores it to the mailbox. Device Claim allows the Sender device presenting it to read and write data to / from the mailbox, thus binding it to the mailbox.
 
-Sender device sends mailboxIdentifier to the Relay server as a part of CreateMailbox request.
+Sender device sends MailboxIdentifier to the Relay server as a part of CreateMailbox request.
 Once mailbox is created, it has limited time to live. When expired, mailbox shall be deleted - refer to DeleteMailbox endpoint. Default expiration period has to be configured on the Relay server.
 
-Relay server builds a unique URL link to a mailbox (for example, “http://relayserver.com/mailbox/1234567890”) and returns it to the Sender device, which sends the link directly to the Receiver device over communication channel (e.g. SMS, email, iMessage).
-In addition to the URL link, Sender device passes a Secret (e.g. symmetric encryption key) to the Receiver device either through the same channel (in the same message as the URL) or over a different channel.
+Relay server builds a unique URL link to a mailbox (for example, “http://relayserver.com/mailbox/1234567890”) and returns it to the Sender device, which sends the link directly to the Receiver device over communication channel (e.g. SMS, email, messaging application).
+In addition to the URL link, Sender device passes a Secret to the Receiver device either through the same channel (in the same message as the URL) or over a different channel. Please refer to section "Security Considerations" for more details.
 
-Receiver device, having obtained both the URL link and the Secret,  generates a unique token - Receiver Device Claim - and passes it to the Relay server in order to read the encrypted Provisioning Information from the mailbox. 
+Receiver device, having obtained both the URL link and the Secret, generates a unique token - Receiver Device Claim - and passes it to the Relay server in order to read the encrypted Provisioning Information from the mailbox. 
```

Comment:
The receiver device can bind to the mailbox before getting the secret. I'm not sure if it is worth making that distinction here, but wanted to call it out.

Reply:
Receiver device claims the mailbox at the time it makes a call to read secure content. 
Reading OpenGraph display info doesn't require a claim token